src/transformers/models/bartpho/tokenization_bartpho.py

```diff
@@ -151,27 +147,30 @@ def __init__(
             **kwargs,
         )
 
-        self.vocab_file = vocab_file
-        self.monolingual_vocab_file = monolingual_vocab_file
```

Comment:
The "new" spm model, producing the same tokenization output as the "old" 250K-sized model, now can represent the monolingual Vietnamese vocabulary without the `monolingual_vocab_file` file.

For example, the following two functions return the same results.

```
def _convert_id_to_token(self, index):
         """Converts an index (integer) in a token (str) using the vocab."""
         if index in self.fairseq_ids_to_tokens:
             return self.fairseq_ids_to_tokens[index]
         return self.sp_model.IdToPiece(index - self.fairseq_offset)
         
def _convert_id_to_token(self, index):
         """Converts an index (integer) in a token (str) using the vocab."""
         return self.fairseq_ids_to_tokens[index]
 ```

Reply:
For my part, I would just like to point out why I think it is complicated to have a tokenizer fast equivalent to slow (proposed in your PR #13788) and why I think the new version is not equivalent to the one proposed before.

What's special about your slow tokenizer is that you tokenize plus you set ids. If the id does not belong to your monolingual vocab then you replace it with the id of the unknown token. This logic is implemented here: 
https://github.com/huggingface/transformers/blob/ade7371a41145c53fe90f66da7e5606f96068e98/src/transformers/models/bartpho/tokenization_bartpho.py#L284-L289

This logic has a particular effect when the input text contains tokens present in the `vocab_file` but not in the `monolingual_vocab_file`.

Indeed, with the version of transformers on master:
```python
from transformers import BartphoTokenizer

model_name = "vinai/bartpho-syllable"
tokenizer_s = BartphoTokenizer.from_pretrained(model_name, revision="e237566c590483a1bdd03aba4687d7516d091e0f")

text = " 접수居家"
tokenizer_s.convert_ids_to_tokens(tokenizer_s.encode(text))
```
gives
```python
['<s>', '<unk>', '<unk>', '</s>']
```

Note: I apologize if VV means something weird, I just concatenated 2 tokens to illustrate my previous point. I have no idea what it means (or doesn't mean).

With the new fast version you propose, here is what it would look like:
```python
from transformers import BartphoTokenizerFast

model_name = "vinai/bartpho-syllable"
tokenizer_f = BartphoTokenizerFast.from_pretrained(model_name)

text = " 접수居家"
tokenizer_f.convert_ids_to_tokens(tokenizer_f.encode(text))
```
gives
```python
['<s>', '▁', '<unk>', '수', '居', '家', '</s>']
```

I hope it makes sense! :smile:

Reply:
@SaulLu I realized this. To develop the fast tokenizer (tokenizer_file), I had to hack into the pre-trained XLM-R sentencepiece model to reduce its vocab size from 250k to 40k (this is for aligning with the existing monolingual vocabulary). Thus, it might affect the BPE subword segmentation/tokenization.

In particular, on [a monolingual Vietnamese corpus](https://drive.google.com/file/d/1wyOW6dXM2iphVhj-8K0LU_XAQPuETkM5/view?usp=sharing) of 590K sentences (~18M tokens, i.e. the averaged sentence length is **30** tokens), the fast and the original slow tokenizers produce different subword tokenization outputs for only 14.5K sentences (rate: 14.5K / 590K ~ **1 / 40**). Note that the difference in subword tokenization mismatch, in most cases, is accounted for only one token (genearlly, less appearing token) in each of the 14K sentences. So the token-level rate for appearing a subword tokenization mismatch between the fast and slow tokenizers is about 1 / (40 * 30) = 1 / 1200, very minor, not really affecting model's performance, I believe.

Reply:
```python
count1, count2 = 0, 0
with open("Vietnamese_file_path", "r") as infile:
    for line in infile:
        count1 += 1

        tokenIDs_f = tokenizer_f(line.strip())['input_ids']
        tokenIDs_s = tokenizer_s(line.strip())['input_ids']
        
        if tokenIDs_f != tokenIDs_s:
            count2 += 1
```

Reply:
Btw, the text in your example is Chinese, so it is not surprising that you observed the subword tokenization mismatch. @SaulLu

Reply:
The same very minor mismatch rate for BPE subword tokenization is also observed on [another Vietnamese dataset](https://drive.google.com/file/d/1mhp0BMNkZq9SBxeNYrJx08BeNRerzpnl/view?usp=sharing) of about 9M sentences. @SaulLu 

> @SaulLu I realized this. To develop the fast tokenizer (tokenizer_file), I had to hack into the pre-trained XLM-R sentencepiece model to reduce its vocab size from 250k to 40k (this is for aligning with the existing monolingual vocabulary). Thus, it might affect the BPE subword segmentation/tokenization.
> 
> In particular, on [a monolingual Vietnamese corpus](https://drive.google.com/file/d/1wyOW6dXM2iphVhj-8K0LU_XAQPuETkM5/view?usp=sharing) of 590K sentences (~18M tokens, i.e. the averaged sentence length is **30** tokens), the fast and the original slow tokenizers produce different subword tokenization outputs for only 14.5K sentences (sentence-level mismatch rate: 14.5K / 590K ~ **1 / 40**). Note that the difference in subword tokenization mismatch, in most cases, is accounted for only one token (genearlly, less appearing token) in each of the 14K sentences. So the token-level rate for appearing a subword tokenization mismatch between the fast and slow tokenizers is about 1 / (40 * 30) = 1 / 1200, very minor, not affecting model's performance, I believe.

Reply:
I think the tiny different rate between tokenization outputs of the original slow and new fast tokenizers confirms that the new fast tokenizer is equivalent to the original slow one. 
I am wondering what your thought is w.r.t. this? Thank you @SaulLu